src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -2890,12 +2890,11 @@ public RubyNode visitSplatNode(SplatParseNode node) {
     public RubyNode visitStrNode(StrParseNode node) {
         final SourceIndexLength sourceSection = node.getPosition();
 
-        final Rope nodeRope = node.getValue();
-        final CodeRange codeRange = node.getCodeRange();
-        final Rope rope = context.getRopeCache().getRope(nodeRope, codeRange);
-
+        final Rope rope = context.getRopeCache().getRope(node.getValue(), node.getCodeRange());
         final RubyNode ret;
 
+        // isFrozen() is set with the magic frozen_string_literal comment or the command line flag.
+        // For the command line flag, we do not want to apply it to the core library files.
```

Comment:
I think this comment is missing a detail. How are we delineating between the froze_string_literal comment case and the command line flag case? Otherwise, the predicate following this comment looks wrong since it rejects files in core in all cases.

Reply:
Okay. But should there be a TODO or caveat added to the comment?

Reply:
I added a TODO.